src/sentry/app.py

```diff
@@ -1,13 +1,3 @@
-from threading import local
-
-
-class State(local):
-    request = None
-    data = {}
-
-
-env = State()
-
```

Comment:
A bit nervous about touching what looks like a core file, but all I'm trying to do here is avoid the import chain starting on line 18.

`sentry.app.env`'s usage in `src/sentry/services/hybrid_cloud/auth/__init__.py` fails because of all the circular imports in that chain, so I moved the env to it's own file to attempt to resolve this and imported it directly.

Reply:
Also not an expert in python modules but I believe all imports following:

```python
from sentry.app import env
```

should still be satisfied. I've opted not to change any of those behaviors in this PR aside from the hybrid cloud one I'm addressing.

Reply:
You are correct in that assessment.  I'm still wondering if this will all be fixed if I just mix my signal connecting code issue, hold on.